cmd/opentelemetry/app/receiver/kafkareceiver/kafka_receiver.go

```diff
@@ -17,79 +17,135 @@ package kafkareceiver
+			Password:    opts.Kerberos.Password,
+			ConfigPath:  opts.Kerberos.ConfigPath,
+			KeyTabPath:  opts.Kerberos.KeyTabPath,
+		}
+	}
+
+	if opts.Authentication == "plaintext" {
+		cfg.Authentication.PlainText = &kafkaexporter.PlainTextConfig{
+			Username: opts.PlainText.UserName,
+			Password: opts.PlainText.Password,
+		}
+	}
+
+	if opts.Authentication == "tls" && opts.TLS.Enabled {
+		cfg.Authentication.TLS = &configtls.TLSClientSetting{
+			TLSSetting: configtls.TLSSetting{
+				CAFile:   opts.TLS.CAPath,
+				CertFile: opts.TLS.CertPath,
+				KeyFile:  opts.TLS.KeyPath,
+			},
+			ServerName: opts.TLS.ServerName,
+			Insecure:   opts.TLS.SkipHostVerify,
+		}
+	}
+
+	return cfg
+}
+
+// CreateTraceReceiver creates Jaeger receiver trace receiver.
+// This function implements OTEL component.ReceiverFactory interface.
+func (f *Factory) CreateTraceReceiver(
+	ctx context.Context,
 	params component.ReceiverCreateParams,
+	cfg configmodels.Receiver,
+	nextConsumer consumer.TraceConsumer,
 ) (component.TraceReceiver, error) {
-	w := &writer{receiver: config.Name(), nextConsumer: nextConsumer}
-	consumer, err := builder.CreateConsumer(
-		params.Logger,
-		metrics.NullFactory,
-		w,
-		config.Options)
-	if err != nil {
-		return nil, err
-	}
-	return &kafkaReceiver{
-		consumer: consumer,
-		logger:   params.Logger,
-	}, nil
+	return f.Wrapped.CreateTraceReceiver(ctx, params, cfg, nextConsumer)
 }
 
-// Start starts the receiver.
-func (r kafkaReceiver) Start(_ context.Context, _ component.Host) error {
-	r.consumer.Start()
-	return nil
+// CreateMetricsReceiver creates a metrics receiver based on provided config.
+// This function implements component.ReceiverFactory.
+func (f *Factory) CreateMetricsReceiver(
+	ctx context.Context,
+	params component.ReceiverCreateParams,
+	cfg configmodels.Receiver,
+	nextConsumer consumer.MetricsConsumer,
+) (component.MetricsReceiver, error) {
+	return f.Wrapped.CreateMetricsReceiver(ctx, params, cfg, nextConsumer)
 }
 
-// Shutdown shutdowns the receiver.
-func (r kafkaReceiver) Shutdown(_ context.Context) error {
-	return r.consumer.Close()
+// CreateLogsReceiver creates a receiver based on the config.
+// If the receiver type does not support logs or if the config is not valid
+// error will be returned instead.
+func (f Factory) CreateLogsReceiver(
+	ctx context.Context,
+	params component.ReceiverCreateParams,
+	cfg configmodels.Receiver,
+	nextConsumer consumer.LogsConsumer,
+) (component.LogsReceiver, error) {
+	return f.Wrapped.CreateLogsReceiver(ctx, params, cfg, nextConsumer)
 }
 
-// WriteSpan writes a span to the next consumer.
-func (w writer) WriteSpan(ctx context.Context, span *model.Span) error {
-	batch := model.Batch{
-		Spans:   []*model.Span{span},
-		Process: span.Process,
+// MustOtelEncodingForJaegerEncoding translates a jaeger encoding to a otel encoding
+func MustOtelEncodingForJaegerEncoding(jaegerEncoding string) string {
+	switch jaegerEncoding {
```

Comment:
missing `kafka.EncodingZipkinThrift` ?

Reply:
We should also support all OTEL's encodings.

Reply:
> We should also support all OTEL's encodings.
This method is used to translate jaeger config/cli to OTEL config.  Do we want to support specifying otel encodings in Jaeger config?  If so I will add them, just want to be clear.

> missing kafka.EncodingZipkinThrift ?
[OTEL Kafka Exporter](https://github.com/open-telemetry/opentelemetry-collector/tree/master/exporter/kafkaexporter) does not support zipkin encodings.  Only the receiver does.  I think we need both otel receiver and exporter to support zipkin for Jaeger to support it as an option.

Reply:
>  Do we want to support specifying otel encodings in Jaeger config

In the Jaeger config we can support only already supported encodings.